tests/foreman/cli/test_webhook.py

```diff
@@ -0,0 +1,216 @@
+        options = options or {}
+        if not any(options.get(key) for key in ['organizations', 'organization-ids']):
+            options['organization-id'] = org.id
+        if not any(options.get(key) for key in ['locations', 'locations-ids']):
+            options['location-id'] = loc.id
+        if options.get('name') is None:
+            options['name'] = gen_alphanumeric()
+        if options.get('event') is None:
+            options['event'] = choice(WEBHOOK_EVENTS)
+        if options.get('http-method') is None:
+            options['http-method'] = choice(WEBHOOK_METHODS)
+        if options.get('target-url') is None:
+            options['target-url'] = 'http://localhost/some-path'
+
+        return Box(Webhook.create(options))
+
+    return partial(_create_webhook, org=class_org, loc=class_location)
+
+
+def assert_created(options, hook):
+    for option in options.items():
+        if not option[0] in ['event', 'organization-id', 'location-id']:
+            assert hook[option[0]] == option[1]
+
+
+class TestWebhook:
+    @pytest.mark.tier2
+    def test_negative_invalid_event(self, webhook_factory):
+        """Test negative webhook creation with an invalid event
+
+        :id: 60cd456a-9943-45cb-a72e-23a83a691499
+
+        :expectedresults: Webhook is not created
+
+        :CaseImportance: High
+        """
+        with pytest.raises(CLIError):
+            webhook_factory(options={'event': 'invalid_event'})
+
+    @pytest.mark.tier2
+    @pytest.mark.parametrize('event', **parametrized(WEBHOOK_EVENTS))
+    def test_positive_valid_event(self, event, webhook_factory):
+        """Test positive webhook creation with a valid event
+
+        :id: 9b505f1b-7ee1-4362-b44c-f3107d043a05
+
+        :expectedresults: Webhook is created with event specified
+
+        :CaseImportance: High
+        """
+        hook = webhook_factory(options={'event': event})
+        assert event in hook.event
+
+    @pytest.mark.tier2
+    def test_negative_invalid_method(self, webhook_factory):
+        """Test negative webhook creation with an invalid HTTP method
+
+        :id: 573be312-7bf3-4d9e-aca1-e5cac810d04b
+
+        :expectedresults: Webhook is not created
+
+        :CaseImportance: High
+        """
+        with pytest.raises(CLIError):
+            webhook_factory(options={'http-method': 'NONE'})
+
+    @pytest.mark.tier2
+    @pytest.mark.parametrize('method', **parametrized(WEBHOOK_METHODS))
+    def test_positive_valid_method(self, method, webhook_factory):
+        """Test positive webhook creation with a valid HTTP method
+
+        :id: cf8f276a-d21e-44d0-92f2-657232240c7e
+
+        :expectedresults: Webhook is created with specified method
+
+        :CaseImportance: High
+        """
+        hook = webhook_factory(options={'http-method': method})
+        assert hook['http-method'] == method
+
+    @pytest.mark.tier1
+    def test_positive_update_hook(self, webhook_factory):
```

Comment:
Isn't this covered in `test_positive_end_to_end`?, do we need a separate test for it?

Reply:
That's a good question. I wasn't sure so I added some small tests, and figured this way debugging would be easier if there's ever a failure. @mshriver @JacobCallahan what do you think?

Reply:
We have been doing this in the past, but it led to gradually bloating the test suite, I'd vote against duplication. Anything failing in atomic tests would fail in end to end as well.

Reply:
I'd agree here with consolidating the test functions if the individual ones are not providing coverage of the specific action beyond what the E2E test contains. These aren't parametrized to provide additional coverage.